spec/system/docs_index_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+describe "Discourse Docs | Index", type: :system do
+  fab!(:current_user) { Fabricate(:user) }
+  fab!(:category) { Fabricate(:category) }
+  fab!(:topic_1) { Fabricate(:topic, category: category) }
+  fab!(:topic_2) { Fabricate(:topic, category: category) }
+  fab!(:post_1) { Fabricate(:post, topic: topic_1) }
+  fab!(:post_2) { Fabricate(:post, topic: topic_2) }
+
+  before do
+    SiteSetting.docs_enabled = true
+    SiteSetting.docs_categories = category.id.to_s
+    sign_in(current_user)
+  end
+
+  it "does not error when showing the index" do
+    visit("/docs")
+    expect(page).to have_css(".docs-topic-link", text: topic_1.title)
+    expect(page).to have_css(".docs-topic-link", text: topic_2.title)
+  end
+
+  describe "topic excerpts" do
+    before do
+      topic_1.update_excerpt(post_1.excerpt_for_topic)
+      topic_2.update_excerpt(post_2.excerpt_for_topic)
+    end
+
+    it "does not show the topic excerpts by default" do
+      visit("/docs")
+      expect(page).to have_no_css(".topic-excerpt")
+    end
+
+    context "when docs_show_topic_excerpts is true" do
+      before do
+        SiteSetting.docs_show_topic_excerpts = true
+        SiteSetting.always_include_topic_excerpts = true
```

Comment:
@martin-brennan we should probably set this up so that the hidden core `always_include_topic_excerpts` isn't requierd for this to work. 

Probably have a way for docs to hook into excerpts in the serializer, a bit like the theme modifier stuff here: 

https://github.com/discourse/discourse/blob/main/app/serializers/listable_topic_serializer.rb#L136-L140

Reply:
It's not -- the theme modifier `serialize_topic_excerpts` can be used, and that was what Jordan was trying to use originally. I only used `always_include_topic_excerpts` in the test because it was easier. This PR explains it https://github.com/discourse/discourse-docs/pull/139 , the original issue was that the theme modifier wasn't working in the docs plugin because we were re-initializing Guardian.

I didn't realise that @jordanvidrine opened this one which incorporates the work from that PR. We should  also incorporate my description into the commit message/PR description, will do that now.

Reply:
I can try updating the test here too to use a theme modifier instead so it's clearer if that helps.

Reply:
> when enabled and used in conjunction either with the always_include_topic_excerpts site setting or the serialize_topic_excerpts theme modifier.

Yeah, I see. But, from an "admin looking at site settings" perspective, this is going to be confusing. Enabling `docs_show_topic_excerpts` will not enable excerpts unless one of those two other conditions is met. (And they are tricky conditions for admins to know about, `always_include_topic_excerpts` is hidden site settings, theme modifiers are quite technical.) 

Do we need the `docs_show_topic_excerpts` setting at all? Can we just have docs output an excerpt if it's present in the serializer?

Reply:
A couple of other options: 

- add a "show in docs" setting to the excerpts theme component
- make the `docs_show_topic_excerpts` setting in docs hidden (doing so, would avoid having to support admins who see that setting and when they enable it say "why is this not working"?)

Reply:
Option 1 should be relatively straightforward.